.github/workflows/build-waypoint-oss.yml

```diff
@@ -0,0 +1,123 @@
+      package-name:
+        type: string
+        required: true
+      waypoint-version:
+        type: string
+        required: true
+      waypoint-base-version:
+        type: string
+        required: true
+      waypoint-prerelease:
+        type: string
+        required: true
+      waypoint-metadata:
+        type: string
+        required: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    outputs:
+      bin-name: ${{ steps.get-bin-name.outputs.bin-name }}
+    name: Build ${{ inputs.package-name }} ${{ inputs.goos }} ${{ inputs.goarch }} v${{ inputs.waypoint-version }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Restore ui assets
+        uses: actions/cache/restore@v3
+        with:
+          path: ui/dist/assets
+          key: ui_assets_${{ github.sha }}
+          fail-on-cache-miss: true
+          enableCrossOsArchive: true
+      - name: Restore build assets
+        uses: actions/cache/restore@v3
+        with:
+          path: internal/assets
+          key: build_assets_${{ github.sha }}
+          fail-on-cache-miss: true
+          enableCrossOsArchive: true
+      - name: Determine binary name
+        id: get-bin-name
+        run: |
+          bin_name="${{ inputs.package-name }}"
+          if [ "${{ inputs.goos }}" == "windows" ];then
+               bin_name="${bin_name}.exe"
+          fi
+          echo "bin-name=${bin_name}" | tee -a "${GITHUB_OUTPUT}"
+      - name: Build binary
+        env:
+          GOOS: ${{ inputs.goos }}
+          GOARCH: ${{ inputs.goarch }}
+          WAYPOINT_GOOS: ${{ inputs.goos }}
+          WAYPOINT_GOARCH: ${{ inputs.goarch }}
+          BASE_VERSION: ${{ inputs.waypoint-base-version }}
+          PRERELEASE: ${{ inputs.waypoint-prerelease }}
+          ACTIONSOS: ${{ inputs.runson }}
+          CGO_ENABLED: ${{ inputs.cgo-enabled }}
+          CRT_BIN_NAME: ${{ steps.get-bin-name.outputs.bin-name }}
+        uses: hashicorp/actions-go-build@v0.1.7
+        with:
+          product_name: ${{ inputs.package-name }}
+          product_version: ${{ inputs.waypoint-version }}
+          go_version: ${{ inputs.go-version }}
+          os: ${{ inputs.goos }}
+          arch: ${{ inputs.goarch }}
+          reproducible: nope
+          instructions: |
+            make bin/crt-${{ inputs.package-name }}
+      - name: Linux packaging
+        if: ${{ (inputs.goos == 'linux') && (inputs.package-name == 'waypoint') && (inputs.goarch != '386') }}
+        uses: hashicorp/actions-packaging-linux@v1
+        with:
+          name: ${{ inputs.package-name }}
+          description: ${{ inputs.package-name }}
+          arch: ${{ inputs.goarch }}
+          version: ${{ inputs.waypoint-version }}+${{ inputs.waypoint-metadata }}
+          maintainer: "HashiCorp"
+          homepage: "https://github.com/hashicorp/waypoint"
+          license: "MPL-2.0"
+          binary: "dist/${{ inputs.package-name }}"
+          deb_depends: "git"
+          rpm_depends: "git"
```

Comment:
Is this an actual runtime dependency of waypoint?

Reply:
We do use submodules regarding 3rd party protocol buffers and run a few `git` commands in other makefile steps, would those reasons explain this need?

Reply:
100%

Reply:
> Do end-users need git installed to use waypoint?

No